scipy/sparse/construct.py

```diff
@@ -404,19 +404,25 @@ def _compressed_sparse_stack(blocks, axis):
     """
     other_axis = 1 if axis == 0 else 0
     data = np.concatenate([b.data for b in blocks])
-    indices = np.concatenate([b.indices for b in blocks])
-    indptr = []
-    last_indptr = 0
     constant_dim = blocks[0].shape[other_axis]
+    idx_dtype = get_index_dtype(arrays=[b.indptr for b in blocks],
+                                maxval=max(data.size, constant_dim))
+    indices = np.empty(data.size, dtype=idx_dtype)
+    indptr = np.empty(sum(b.shape[axis] for b in blocks) + 1, dtype=idx_dtype)
+    last_indptr = 0
```

Comment:
One final thing --- I was wrong that `last_indptr = idx_dtype(0)` is not necessary, it is necessary.

The Python integer is cast to the default integer size, but on win64 that's 32-bit even when the address space is 64-bit.